Detect saliency layer (#583)
ServiceNow/azimuth: docs/docs/reference/configuration/model_contract.md

    2. List of pipelines. Can also be set to `null` to launch Azimuth with a dataset only.
    3. Enable uncertainty quantification.
    4. Layer name where to calculate the gradients, normally the word embeddings layer.

>        Only available for Pytorch models. Defaults to "auto"; in that case the layer will be found using
>        `model.base_model.get_input_embeddings()`.

Reviewer: If that makes sense
```suggestion
       Only available for Pytorch models. Defaults to "auto", in which case the embedding layer will be detected automatically using `model.base_model.get_input_embeddings()`.
```

## Saliency Layer

🟡 **Default value**: `auto`

If using a Pytorch model, [:material-link: Saliency Maps](../../key-concepts/saliency.md) can be

> available. Specify the name of the embedding layer on which to compute them. The default is set to "auto".
> In that case, the embedding layer will be found using `model.base_model.get_input_embeddings()`.

Reviewer: Detail, but I would suggest
```suggestion
available. Specify the name of the embedding layer on which to compute them. The default is set to "auto",
in which case the embedding layer will be detected automatically using `model.base_model.get_input_embeddings()`.
```

Reviewer: So, if I don't have the necessary layer in my model, I now have to specify `"saliency_layer": null`, whereas that was the default before. Is it worth noting the breaking change in the change log? Or would there be a way to also support that use case with the `"auto"`?

Author: "auto" will also work when your model is not a HF pipeline (and will be equivalent to "null"). @Dref360, do you foresee a use case where the model would be a HF pipeline, but the `input_embedding` layer wouldn't be appropriate for saliency_maps?

Reviewer 2: Maybe in a bizarre use case with multiple inputs, but this is not a supported use case.

Author: Thanks. Then I don't think it's a breaking change, but I added it to the changelog as a new feature. LMK what you think.
